sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -436,23 +421,19 @@ public void processElement(ProcessContext context) throws Exception {
       @FinishBundle
       public void finishBundle(Context context) throws Exception {
         for (T record : batch) {
-          try {
-            PreparedStatement statement = connection.prepareStatement(query);
-            preparedStatementSetter.setParameters(record, statement);
-            try {
-              statement.executeUpdate();
-            } finally {
-              statement.close();
-            }
-          } catch (Exception e) {
-            LOGGER.error("Can't insert into database", e);
-          }
+          preparedStatementSetter.setParameters(record, preparedStatement);
+          preparedStatement.executeUpdate();
```

Comment:
In that case, I don't need the `batch` list anymore: just `batchSize`.
In the `processElement` method, I can do:
```
T record = context.element();
preparedStatement.clearParameters();
preparedStatementSetter.setParameters(record, preparedStatement);
preparedStatement.addBatch();
batchCount++;
if (batchCount >= batchSize) {
  finishBundle(context);
}
```
Then, in the `finishBundle`, I do:
```
preparedStatement.executeBatch();
```
Agreed ?